superset/utils/log.py

```diff
@@ -118,12 +118,23 @@ def log_with_context(  # pylint: disable=too-many-locals
 
         duration_ms = int(duration.total_seconds() * 1000) if duration else None
 
+        # Initial try and grab user_id via flask.g.user
         try:
             user_id = g.user.get_id()
         except Exception as ex:  # pylint: disable=broad-except
-            logging.warning(ex)
             user_id = None
 
+        # Whenever a user is not bounded to a session we
+        # need to add them back before logging to capture user_id
+        if user_id is None:
+            try:
+                session = current_app.appbuilder.get_session
+                session.add(g.user)
```

Comment:
@etr2460 At this point in time the `g.user` is available the problem is when we try and grab the `user_id`. Not sure boss why get_id() is cause the unbounding session error. 

I've been working on this for a week and this is working properly in adding the user back to session and allowing us to query the user_id for logging.

Reply:
weird.... @dpgaspar do you have any ideas why this might happen?

If this is the only way to fix it, it's probably fine, but this was pretty confusing to me